app/code/Magento/Customer/Model/Session.php

```diff
@@ -341,7 +341,7 @@ public function setCustomerId($id)
     public function getCustomerId()
     {
         if ($this->storage->getData('customer_id')) {
-            return $this->storage->getData('customer_id');
+            return (int) $this->storage->getData('customer_id');
```

Comment:
This changes affects failed tests, this method should return null or int in your case it always return int, because (int)null == 0;

Reply:
Here is `if ($this->storage->getData('customer_id')) {` one line above, it should return null if the value is null.

Reply:
ah, i see agree